site/src/pages/_document.page.tsx

```diff
@@ -92,6 +88,11 @@ class MyDocument extends Document {
         <body>
           <Main />
           <NextScript />
+          <script
+            defer
+            src="https://kit.fontawesome.com/31605611a9.js"
+            crossOrigin="anonymous"
+          />
```

Comment:
We should probably change how we use font awesome because the current one changes the DOM outside React. This may lead to crashes because React reconciler expects that it ‘owns’ the tree.

Here’s what the DOM looks like at the moment:

<img width="1361" alt="Screenshot 2022-01-04 at 23 20 54" src="https://user-images.githubusercontent.com/608862/148138660-6a9d4d83-89cf-491d-89e6-6df53cb0fc79.png">

The comment is a Node that was created by React but then removed by a third-party Font-awesome script. My fix is to do with delaying this dodgy behaviour till the tree is re-hydrated, but the approach is still fundamentally wrong.

Reply:
Understood, happy to do it like this for. I've come across this before and didn't think of that solution. The reason I've left this unaddressed because I think we should change how we load the font awesome icons.

As I understand it, currently the SVG icons are loaded and added to the DOM by the front awesome script on the client side. If I'm correct, this would explain why the icons currently aren't instantly loading (they appear a couple milliseconds after the rest page has loaded).

Ideally we want the SVG icons to be attached to the document when the page is built on the server by Next.js instead, so that they can be provided to the browser directly (instead of relying on the browser to fetch them). I haven't tested this out, but I think we could accomplish this by using the font awesome NPM packages (such as `@fortawesome/fontawesome-svg-core`) to import the svgs directly.

I haven't spent a lot of time researching/testing this further as I've considered it a lower priority item - nonetheless I think we should do this at some point soon, so that the way we are using icons in development of the FE is more future-proof.

Reply:
Yeah I think we are aligned here. We don’t have to change the approach immediately, but worth being cautious about what the non-react script does.

`@fortawesome/fontawesome-svg-core` and `@fortawesome/react-fontawesome` should be the right way to go. I would not worry too much about adding more icons before the switchover, because we can bulk-replace them with Search later on. Still, it's something we want to do within weeks than months.